packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -483,6 +485,16 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
             }
           }
 
+          s.append(`\
+const __viteFileDeps = ${JSON.stringify(
+  fileDeps.map((fileDep) =>
+    relativePreloadUrls ? path.relative(path.dirname(file), fileDep) : fileDep
+  )
+)}
+function __viteMapDep(indexes) {
+  return indexes.map((i) => __viteFileDeps[i])
+}`)
```

Comment:
Looks like some of our tests invokes `__viteMapDep` on initialize, causing `ReferenceError: Cannot access uninitialized variable.` in some cases. We might have to move `__viteFileDeps` back in the function like before, or prepend this code instead.

Also this file needs to run `pnpm format`.

Reply:
Moved it back into `__viteMapDep` and applied `npnm format`.

Reply:
Can't we prepend it instead? I am still not so sure about having it nested. Even JS engine might optimize it, it still needs to create new references and free the memory on each run.

Reply:
Hmm yeah I guess we could try prepend first. Thought it might have perf implications since we're prepending in a sourcemap, but I don't think it would matter actually.

Reply:
Another trick I can think about is:

```ts
function __viteMapDep(indexes) {
  return indexes.map((i) => this.files[i])
}
__viteMapDep.files = JSON.stringify({})
```